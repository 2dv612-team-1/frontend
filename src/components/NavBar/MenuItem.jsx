import styled from "styled-components";
import { Link } from "react-router-dom";
import colors from "../../constants/colors";

const MenuItem = styled(Link)`
  margin: 0 10px 0 10px;
  font-size: 18px;
  color: ${colors.secondary};
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`;

export default MenuItem;
