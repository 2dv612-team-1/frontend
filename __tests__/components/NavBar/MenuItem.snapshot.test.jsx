import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter as Router } from "react-router-dom";
import "jest-styled-components";
import MenuItem from "../../../src/components/NavBar/MenuItem";

describe("MenuItem", () => {
  let props;
  let wrapper;

  const Wrapper = () => {
    if (!wrapper) {
      wrapper = shallow(
        <Router>
          <MenuItem {...props}>{props.children}</MenuItem>
        </Router>
      );
    }
    return wrapper;
  };

  // Reset Component and Props before each test.
  beforeEach(() => {
    props = {
      children: undefined
    };
    wrapper = undefined;
  });

  describe("when props are undefined", () => {
    it("Should match snapshot", () => {
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });

  describe("when props are defined", () => {
    it("Should match snapshot", () => {
      props.children = "Text";
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });
});
