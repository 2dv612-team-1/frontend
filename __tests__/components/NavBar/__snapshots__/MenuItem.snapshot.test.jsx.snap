// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem when props are defined Should match snapshot 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Styled(Link)>
    Text
  </Styled(Link)>
</Router>
`;

exports[`MenuItem when props are undefined Should match snapshot 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Styled(Link) />
</Router>
`;
